src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,289 @@
+"""
+Define a Rotation Class for customized rotation operations used by ASPIRE.
+"""
+
+import copy
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import svd
+from scipy.spatial.transform import Rotation as sp_rot
+
+from aspire.utils import ensure
+from aspire.utils.matlab_compat import Random
+
+
+class Rotation:
+    """
+    Define a Customized Rotation class
+    """
+
+    def __init__(self, num_rots, seed=None, seq='ZYZ',
+                 angles=None, degrees=False, dtype=np.float32):
+        """
+        Initialize a Rotation object
+
+        :param num_rots: Total number of rotation sets
+        :param seed: Seed for initializing random rotations
+        :param seq:  Sequence of order applying Euler angles
+        :param angles: Euler angles to generate rotation matrices
+        :param degrees: Whether Euler angles in degrees or not
```

Comment:
if False, then what?

Reply:
Since our application cases  in `ImageSource` class use degrees, I have removed this option and well will use degrees directly in new commit 9f6e3b3.